src/jwt_authorizer/config.py

```diff
@@ -2,140 +2,300 @@
+        """Construction a Config object from Dynaconf settings.
 
         Parameters
         ----------
-        app : `flask.Flask`
-            The Flask application to configure.
-        user_config : `str`
-            An additional configuration file to load.
+        settings : `dynaconf.LazySettings`
+            Dynaconf settings.
+
+        Returns
+        -------
+        config : `Config`
+            The corresponding Config object.
         """
-        global logger
-        defaults_file = os.path.join(
-            os.path.dirname(__file__), "defaults.yaml"
+        if settings.get("OAUTH2_JWT.KEY"):
+            key = settings["OAUTH2_JWT.KEY"]
+        else:
+            key = cls._load_secret(settings["OAUTH2_JWT.KEY_FILE"])
+        issuer_config = IssuerConfig(
+            iss=settings["OAUTH2_JWT.ISS"],
+            kid=settings["OAUTH2_JWT.KEY_ID"],
+            aud=settings["OAUTH2_JWT.AUD.DEFAULT"],
+            aud_internal=settings["OAUTH2_JWT.AUD.INTERNAL"],
+            key=key,
+            exp_minutes=settings["OAUTH2_JWT_EXP"],
         )
 
-        settings_module = f"{defaults_file},{user_config}"
-        print(settings_module)
-        config = FlaskDynaconf(app, SETTINGS_FILE_FOR_DYNACONF=settings_module)
-        settings = config.settings
-        settings.validators.register(
-            Validator("NO_VERIFY", "NO_AUTHORIZE", is_type_of=bool),
-            Validator("GROUP_MAPPING", is_type_of=dict),
+        group_mapping = {}
+        if settings.get("GROUP_MAPPING"):
+            for key, value in settings["GROUP_MAPPING"].items():
+                assert isinstance(key, str), "group_mapping is malformed"
+                assert isinstance(value, list), "group_mapping is malformed"
+                group_mapping[key] = value
+
+        store_session_settings = settings["OAUTH2_STORE_SESSION"]
+        if store_session_settings.get("OAUTH2_PROXY_SECRET"):
+            secret_b64 = store_session_settings["OAUTH2_PROXY_SECRET"]
+        else:
+            secret_b64 = cls._load_secret(
+                store_session_settings["OAUTH2_PROXY_SECRET_FILE"]
+            )
+        secret = base64.urlsafe_b64decode(secret_b64)
+        session_store = SessionStoreConfig(
+            ticket_prefix=store_session_settings["TICKET_PREFIX"],
+            redis_url=store_session_settings["REDIS_URL"],
+            oauth2_proxy_secret=secret,
+        )
+
+        if settings.get("KNOWN_CAPABILITIES"):
+            known_capabilities = settings["KNOWN_CAPABILITIES"]
+        else:
+            known_capabilities = {}
+
+        issuers = {}
+        if settings.get("ISSUERS"):
+            for url, info in settings["ISSUERS"].items():
+                audience_setting = info["AUDIENCE"]
+                if isinstance(audience_setting, str):
+                    audience: Tuple[str, ...] = (audience_setting,)
+                else:
+                    audience = tuple(audience_setting)
```

Comment:
It can be either a string or a list. I'll add a comment to make that clearer. (This is something else that I'll fix properly when redoing the configuration code. It's supported just for backward compatibility.)